aten/src/ATen/native/LinearAlgebraUtils.h

```diff
@@ -251,47 +258,21 @@ static inline std::tuple<Tensor, Tensor, Tensor> _create_U_S_VT(const Tensor& in
   strides[input.dim() - 1] = m;
   strides[input.dim() - 2] = 1;
 
-  Tensor U_empty;
-  if (!input.is_cuda()) {
-    U_empty = at::empty_strided(sizes, strides, input.options());
-  } else {
-    // NB: U_empty is an empty tensor created on the CPU intentionally, because magma_(d/s)gesdd
-    // (which is the driver routine for the divide and conquer SVD operation)
-    // takes in arrays on the CPU as input. This routine is a hybrid CPU-GPU routine that
-    // moves the inputs between devices internally.
-    U_empty = at::empty_strided(sizes, strides, input.options().device(at::kCPU));
-  }
+  Tensor U_empty = at::empty_strided(sizes, strides, input.options().device(usvt_device));
+  U_empty.zero_();
 
   // VT should be a column-major or a batch of column-major matrices
   sizes[input.dim() - 2] = n;
   sizes[input.dim() - 1] = n;
-  strides = at::detail::defaultStrides(sizes);
-  strides[input.dim() - 1] = n;
-  strides[input.dim() - 2] = 1;
-  Tensor VT_empty;
-  if (!input.is_cuda()) {
-    VT_empty = at::empty_strided(sizes, strides, input.options());
-  } else {
-    // NB: VT_empty is an empty tensor created on the CPU intentionally, because magma_(d/s)gesdd
-    // (which is the driver routine for the divide and conquer SVD operation)
-    // takes in arrays on the CPU as input. This routine is a hybrid CPU-GPU routine that
-    // moves the inputs between devices internally.
-    VT_empty = at::empty_strided(sizes, strides, input.options().device(at::kCPU));
-  }
+  // VT should be a row-major or a batch of row-major matrices
+  Tensor VT_empty = at::zeros(sizes, input.options().device(usvt_device));
+  VT_empty.transpose_(-2, -1);
```

Comment:
I think that the code is correct but the comment is wrong. Moreover, it contradicts the comment at line 264, which says `// VT should be a column-major or a batch of column-major matrices`.
`at::zeros` returns a row-major tensor, and `transpose_` turns it into a column-major one, which is what lapack/magma (and I assume cusolver) expects.
See also https://github.com/pytorch/pytorch/issues/45821#issuecomment-707184617 which is a more in-depth explanation on the subject